be/src/exprs/vectorized/case_expr.cpp

```diff
@@ -249,6 +255,49 @@ class VectorizedCaseExpr final : public Expr {
             columns_has_null |= column->has_null();
         }
 
+        // max case size in use SIMD CASE WHEN implements
+        constexpr int max_simd_case_when_size = 8;
+
+        // optimization for no-nullable Arithmetic Type
+        if constexpr (isArithmeticPT<ResultType>) {
+            bool check_could_use_multi_simd_selector =
+                    !columns_has_null && when_columns.size() <= max_simd_case_when_size && _has_else_expr;
+
+            if (check_could_use_multi_simd_selector) {
+                // TODO: avoid unpack const column
+                for (int i = 0; i < then_columns.size(); ++i) {
+                    then_columns[i] = ColumnHelper::unpack_and_duplicate_const_column(size, then_columns[i]);
+                }
+                for (int i = 0; i < when_columns.size(); ++i) {
+                    when_columns[i] = ColumnHelper::unpack_and_duplicate_const_column(size, when_columns[i]);
+                }
+                for (int i = 0; i < when_columns.size(); ++i) {
+                    ColumnHelper::merge_nullable_filter(when_columns[i].get());
+                }
+
+                using ResultContainer = typename RunTimeColumnType<ResultType>::Container;
+
+                ResultContainer* select_list[then_columns.size()];
+                for (int i = 0; i < then_columns.size(); ++i) {
```

Comment:
Assign then_colums.size() a local variable to make loop easy to optimize.

Reply:
I'm already using max_simd_case_when_size to limit the maximum number of case when

Reply:
Good idea